src/coin.cr

```diff
@@ -226,6 +226,51 @@ class Coin
     end
   end
 
+  # discovers and displays the top five richest currency holders
+  def leaderboard(message)
+    usernames = [] of String
+    bals      = [] of UInt64  
+
+    @redis.keys("*:bal").each do |bal_key|
+      usr_id = bal_key.to_s.split(":")[0].to_u64
+      bal = @redis.get(bal_key)
+      unless bal.nil? || bal == 0
+        usernames << @cache.resolve_user(usr_id).username
+        bals << bal.to_u64
+      end
+    end
+
+    fields = [] of Discord::EmbedField
+    indices = (argsort bals).reverse
+    positions = usernames.size<5 ? usernames.size:5 
+    (0..positions-1).each do |i|
+      fields <<
+        Discord::EmbedField.new(name: "\##{i+1}: #{usernames[indices[i]]}",
+                                value: "Bal: #{bals[indices[i]]}")
+    end
+
+    send_emb message, "", Discord::Embed.new(title: "_Leaderboard:_", fields: fields)
+  end
+
+  # produces a set of indices that correspond to the sorted elements of 'array'
+  def argsort(array)
```

Comment:
since this isn't a class method, and more of a seperate utility, could this be put in its own `utils.cr` file and require'd in?

Reply:
I was just thinking the same thing as I pushed it. Let me fix that.